src/client/Microsoft.Identity.Client/Internal/Requests/ClientCredentialRequest.cs

```diff
@@ -193,6 +200,18 @@ private async Task<MsalAccessTokenCacheItem> GetCachedAccessTokenAsync()
 
             if (cachedAccessTokenItem != null && !_clientParameters.ForceRefresh)
             {
+                if (!string.IsNullOrEmpty(_clientParameters.AccessTokenHashToRefresh))
+                {
+                    string cachedTokenHash = ComputeSHA256(cachedAccessTokenItem.Secret);
+
+                    if (string.Equals(cachedTokenHash, _clientParameters.AccessTokenHashToRefresh, StringComparison.Ordinal))
+                    {
+                        AuthenticationRequestParameters.RequestContext.Logger.Info(
+                            "[ClientCredentialRequest] Found the 'bad' token in the cache. Skipping cache usage.");
```

Comment:
```suggestion
                            "[ClientCredentialRequest] A cached token was found, but it matches the AccessTokenHashToRefresh so it will be skipped");
```